pace-util/pace/util/initialization/allocator.py

```diff
@@ -54,29 +55,52 @@ def from_backend(cls, sizer: SubtileGridSizer, backend: str):
         numpy = StorageNumpy(backend)
         return cls(sizer, numpy)
 
-    def empty(self, dims: Sequence[str], units: str, dtype: type = float):
+    def empty(
+        self,
+        dims: Sequence[str],
+        units: str,
+        dtype: type = float,
+    ):
         return self._allocate(self._numpy.empty, dims, units, dtype)
 
-    def zeros(self, dims: Sequence[str], units: str, dtype: type = float):
+    def zeros(
+        self,
+        dims: Sequence[str],
+        units: str,
+        dtype: type = float,
+    ):
         return self._allocate(self._numpy.zeros, dims, units, dtype)
 
-    def ones(self, dims: Sequence[str], units: str, dtype: type = float):
+    def ones(
+        self,
+        dims: Sequence[str],
+        units: str,
+        dtype: type = float,
+    ):
         return self._allocate(self._numpy.ones, dims, units, dtype)
 
     def _allocate(
-        self, allocator: Callable, dims: Sequence[str], units: str, dtype: type = float
+        self,
+        allocator: Callable,
+        dims: Sequence[str],
+        units: str,
+        dtype: type = float,
     ):
         origin = self._sizer.get_origin(dims)
         extent = self._sizer.get_extent(dims)
         shape = self._sizer.get_shape(dims)
+        mask = tuple(
+            [
+                any(dim in coord_dims for dim in dims)
+                for coord_dims in [X_DIMS, Y_DIMS, Z_DIMS]
+            ]
+        )
+        spatial_dims = [i for j in [X_DIMS, Y_DIMS, Z_DIMS] for i in j]
```

Comment:
I manually tested the code you've written does in fact get the correct spatial_dims, but I still cannot tell from reading this Python list comprehension how it works. This should be more straightforward:
```suggestion
        spatial_dims = X_DIMS + Y_DIMS + Z_DIMS
```